src/cinder/gl/draw.cpp

```diff
@@ -718,6 +833,63 @@ void drawStrokedRect( const Rectf &rect, float lineWidth )
 	ctx->drawArrays( GL_TRIANGLE_STRIP, 0, 16 );
 	ctx->popVao();
 }
+	
+void drawStrokedRoundedRect( const Rectf &r, float cornerRadius, int numSegmentsPerCorner )
+{
+	auto ctx = context();
+	const GlslProg* curGlslProg = ctx->getGlslProg();
+	if( ! curGlslProg ) {
+		CI_LOG_E( "No GLSL program bound" );
+		return;
+	}
+	// automatically determine the number of segments from the circumference
+	if( numSegmentsPerCorner <= 0 ) {
+		numSegmentsPerCorner = (int)math<double>::floor( cornerRadius * M_PI * 2 / 4 );
+	}
+	if( numSegmentsPerCorner < 2 ) numSegmentsPerCorner = 2;
+	
+	GLfloat *verts = new float[(numSegmentsPerCorner+2)*2*4];
+	GLsizei tri = 0;
+	const float angleDelta = 1 / (float)numSegmentsPerCorner * M_PI / 2;
+	const float cornerCenterVerts[8] = {
+		r.x2 - cornerRadius,
+		r.y2 - cornerRadius,
+		r.x1 + cornerRadius,
+		r.y2 - cornerRadius,
+		r.x1 + cornerRadius,
+		r.y1 + cornerRadius,
+		r.x2 - cornerRadius,
+		r.y1 + cornerRadius
+	};
+	for( size_t corner = 0; corner < 4; ++corner ) {
+		float angle = corner * M_PI / 2.0f;
+		vec2 cornerCenter( cornerCenterVerts[corner*2+0], cornerCenterVerts[corner*2+1] );
+		for( int s = 0; s <= numSegmentsPerCorner; s++ ) {
+			vec2 pt( cornerCenter.x + math<float>::cos( angle ) * cornerRadius,
+					cornerCenter.y + math<float>::sin( angle ) * cornerRadius );
+			verts[tri*2+0] = pt.x;
+			verts[tri*2+1] = pt.y;
+			++tri;
+			angle += angleDelta;
+		}
+	}
+	ctx->pushVao();
+	ctx->getDefaultVao()->replacementBindBegin();
+	VboRef defaultVbo = ctx->getDefaultArrayVbo( sizeof(float)*(numSegmentsPerCorner+2)*2*4 );
+	ScopedBuffer bufferBindScp( defaultVbo );
+	defaultVbo->bufferSubData( 0, sizeof(float)*(numSegmentsPerCorner+2)*2*4, verts );
+	
+	int posLoc = curGlslProg->getAttribSemanticLocation( geom::Attrib::POSITION );
+	if( posLoc >= 0 ) {
+		enableVertexAttribArray( posLoc );
+		vertexAttribPointer( posLoc, 2, GL_FLOAT, GL_FALSE, 0, (void*)0 );
+	}
+	ctx->getDefaultVao()->replacementBindEnd();
+	ctx->setDefaultShaderVars();
+	ctx->drawArrays( GL_LINE_LOOP, 0, tri );
+	ctx->popVao();
+	delete [] verts;
```

Comment:
i know this is the from the version in 0.8.6, but could you use unique_ptr, or better yet std::vector, to manage the memory without an explicit delete? You can [reserve()](http://en.cppreference.com/w/cpp/container/vector/reserve) the memory it needs up front to make it the same in efficiency.

Thanks for adding these though, they're one of the last missing ones remaining.